src/openforms/variables/base.py

```diff
@@ -0,0 +1,19 @@
+from dataclasses import dataclass
+
+from openforms.forms.models import FormVariable
+
+
+@dataclass
+class BaseStaticVariable:
+    identifier: str
+
+    def get_initial_value(self, *args, **kwargs):
+        raise NotImplementedError()
+
+    def get_static_variable(self, *args, **kwargs):
+        return FormVariable(
+            name=self.name,
```

Comment:
shouldn't we define `name` and `data_type` as a class property too? You can add them as `dataclasses.field(init=False)` so they are type hinted but not required for the `__init__` method.